examples/mdetr/matcher.py

```diff
@@ -0,0 +1,131 @@
+    matching of the best predictions, while the others are un-matched (and thus treated
+    as non-objects). This implementation is based on the MDETR repo:
+    https://github.com/ashkamath/mdetr/blob/main/models/matcher.py#L13
+
+    Params: cost_class (float): Relative weight of the classification error in the
+                matching cost. Default: 1
+            cost_bbox (float): Relative weight of the L1 error of the bounding box
+                coordinates in the matching cost. Default: 1
+            cost_giou (float): Relative weight of the giou loss of the bounding box in
+                the matching cost. Default: 1
+
+
+    Inputs: pred_logits (Tensor): Classification logits.
+                Size: (batch_size, num_queries, num_classes)
+            pred_boxes (Tensor): Predicted box coordinates.
+                Size: (batch_size, num_queries, 4)
+            target_boxes_per_sample (List[Tensor]): A list of target bounding boxes.
+                Length = batch_size.
+                Each element is a tensor of size (n_boxes_for_sample, 4).
+            positive_map (Tensor): positive_map[i,j] = 1 when box i maps to class j.
+                Size: (total_boxes, num_classes) where total_boxes is the sum of
+                n_boxes_for_sample over every sample in the batch.
+
+    Returns:
+            A list of size batch_size, containing tuples of (index_i, index_j) where:
+                - index_i is the indices of the selected predictions (in order)
+                - index_j is the indices of the corresponding selected targets (in order)
+            For each batch element, it holds:
+                len(index_i) = len(index_j) = min(num_queries, num_target_boxes)
+
+    """
+
+    def __init__(
+        self, cost_class: float = 1, cost_bbox: float = 1, cost_giou: float = 1
+    ):
+        super().__init__()
+        self.cost_class = cost_class
+        self.cost_bbox = cost_bbox
+        self.cost_giou = cost_giou
+        assert (
+            cost_class != 0 or cost_bbox != 0 or cost_giou != 0
+        ), "At least one cost must be nonzero"
+
+    @torch.no_grad()
+    def forward(
+        self,
+        pred_logits: Tensor,
+        pred_boxes: Tensor,
+        target_boxes_per_sample: List[Tensor],
+        positive_map: Tensor,
+    ) -> List[Tuple[Tensor, Tensor]]:
+
+        bs, num_queries = pred_logits.shape[:2]
+        target_boxes = torch.cat(target_boxes_per_sample)
+        # We flatten to compute the cost matrices in a batch
+        out_prob = F.softmax(
+            pred_logits.flatten(0, 1), dim=-1
+        )  # [batch_size * num_queries, num_classes]
+        out_bbox = pred_boxes.flatten(0, 1)  # [batch_size * num_queries, 4]
+        assert target_boxes.size(0) == positive_map.size(
+            0
+        ), "Total of target boxes should match first dim of positive map"
+
+        # Compute the soft-cross entropy between the predicted token alignment and the GT one for each box
```

Comment:
`GT` refers to ? Use the full name here unless it was denoted somewhere else or convention?

Reply:
Oh it means ground truth, but agree that the shortening here is unnecessary